build-system/pr-check/3p-vendor-configs.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Script that builds the 3p vendor configs for AMP runtime during CI.
+ */
+
+const {
+  printSkipMessage,
+  timedExecOrDie,
+  uploadVendorOutput,
+} = require('./utils');
+const {buildTargetsInclude, Targets} = require('./build-targets');
+const {runCiJob} = require('./ci-job');
+
+const jobName = '3p-vendor-configs.js';
+
+/**
+ * @return {void}
+ */
+function pushBuildWorkflow() {
+  timedExecOrDie('gulp update-packages');
+  timedExecOrDie('gulp 3p-vendor-configs --fortesting');
+  uploadVendorOutput();
+}
+
+/**
+ * @return {void}
+ */
+function prBuildWorkflow() {
+  if (
+    buildTargetsInclude(
+      Targets.RUNTIME,
+      Targets.INTEGRATION_TEST,
+      Targets.E2E_TEST,
+      Targets.UNIT_TEST,
+      Targets.VISUAL_DIFF
+    )
+  ) {
+    timedExecOrDie('gulp update-packages');
+    timedExecOrDie('gulp 3p-vendor-configs --fortesting');
+    uploadVendorOutput();
+  } else {
+    printSkipMessage(
+      jobName,
+      'this PR does not affect tests that require building 3p vendor configs'
```

Comment:
For debuggability, I think this line should explicitly spell out ask the jobs that depends on it. This list should be an exact copy of nomodule build.